charts/spire/charts/spire-agent/templates/daemonset.yaml

```diff
@@ -91,11 +91,34 @@ spec:
               elif [ -f "{{ .Values.workloadAttestors.k8s.verification.hostCert.basePath }}/kubelet.crt" ]; then
                 openssl x509 -in "{{ .Values.workloadAttestors.k8s.verification.hostCert.basePath }}/kubelet.crt" -out /hostCert/kubelet.crt
               else
+                {{- if eq .Values.workloadAttestors.k8s.verification.type "auto" }}
+                {{- if $cbh }}
+                URL="https://$NODE_NAME:10250/spec/"
+                {{- else }}
+                URL="https://localhost:10250/spec/"
+                {{- end }}
+                curl --caPath /var/run/secrets/kubernetes.io/serviceaccount/ca.crt "$URL"
+                if [ $? -eq 0 ]; then
+                  echo Mode detected as apiServerCA.
+                  ln -s /var/run/secrets/kubernetes.io/serviceaccount/ca.crt /hostCert/kubelet.crt
+                  exit 0
+                fi
+                {{- end }}
                 echo Could not find certificate.
                 exit 1
               fi
               {{- end }}
+              {{- if eq .Values.workloadAttestors.k8s.verification.type "auto" }}
+              echo Mode detected as hostCert.
+              {{- end }}
```

Comment:
If the mode is detected to host cert does it need to do the checks from lines 81-92?

Reply:
nm. misunderstood code.